src/confirm.rs

```diff
@@ -1,48 +1,230 @@
 use std::time::Duration;
-use futures::{IntoFuture, Future, Stream};
-use api::{Eth, EthFilter, Namespace};
-use types::{U256, TransactionRequest, TransactionReceipt};
+use futures::{IntoFuture, Future, Stream, Poll, Async};
+use futures::stream::Skip;
+use api::{Eth, EthFilter, Namespace, CreateFilter, FilterStream};
+use types::{H256, U256, TransactionRequest, TransactionReceipt};
+use helpers::CallResult;
 use {Transport, Error};
 
-pub fn wait_for_confirmations<'a, T, F, V>(transport: T, poll_interval: Duration, confirmations: usize, check: V)
-  -> Box<Future<Item = (), Error = Error> + 'a> where
-  T: 'a + Transport + Clone,
-  F: 'a + IntoFuture<Item = Option<U256>, Error = Error>,
-  V: 'a + Fn() -> F,
-{
-  let eth = EthFilter::new(transport.clone());
-  let result = eth.create_blocks_filter()
-    .and_then(move |filter| {
-      filter.stream(poll_interval)
-        .skip(confirmations as u64)
-        .and_then(move |_| check())
-        .filter_map(|o| o)
-        .and_then(move |confirmed_block_number| {
-          Eth::new(transport.clone()).block_number()
-            .map(move |last_block_number| confirmed_block_number.low_u64() + confirmations as u64 >= last_block_number.low_u64())
-        })
-        .filter(|confirmed| *confirmed)
-        .take(1)
-        .collect()
-        .map(|_| ())
-    });
-  Box::new(result)
-}
-
-pub fn send_transaction_with_confirmation<'a, T>(transport: T, tx: TransactionRequest, poll_interval: Duration, confirmations: usize) -> Box<Future<Item = TransactionReceipt, Error = Error> + 'a> where
-  T: 'a + Transport + Clone {
-  let eth = Eth::new(transport.clone());
-  let result = eth.send_transaction(tx)
-    .and_then(move |hash| {
-      wait_for_confirmations(transport.clone(), poll_interval, confirmations, move || {
-        eth.transaction_receipt(hash.clone()).map(|option| option.map(|receipt| receipt.block_number))
-      })
-      .and_then(move |_| {
-        let eth = Eth::new(transport.clone());
-        eth.transaction_receipt(hash).and_then(|option| option.ok_or(Error::Unreachable))
+pub trait ConfirmationCheck {
```

Comment:
Should it have a blanket implementation for `T, F where F: Fn() -> T, T: IntoFuture<Item=Option<U256>, Error>`?